x/subscription/spec/02_state.md

```diff
@@ -0,0 +1,52 @@
+<!--
+order: 2
+-->
+
+# State
+
+The `x/subscription` stores the plans and subscriptions on chain:
+
+```golang
+type Plan struct {
+  id int,  // auto-increasing unique identifier
+  title string,
+  description string,
+  owner Address,  // beneficial owner of the plan
+  price Coin,  // price to pay for each period, Coin contains both amount and denomination
+  expiration_time int,  // the timestamp in seconds when plan expires, it will be removed automatically
+  tolerate_failure bool, // if false, it'll automatically cancel subscriptions fails to pay, otherwise merchant can handle them manually
+
+  interval int, // payment interval defined in number of seconds
+  interval_offset int, // interval offset in terms of seconds to support timezones or any arbitrary time offsets
```

Comment:
Can you share a few usecases of `interval_offset` to help me better understand it. Thanks!

Reply:
> Can you share a few usecases of `interval_offset` to help me better understand it. Thanks!

For example:
```
# 0-24 hour in UTC
period index = timestamp / (24 * 3600)
# 0-24 hour in UTC+8
period index = (timestamp + 8 * 3600) / (24 * 3600)

Reply:
Thanks

Reply:
In new revision, it changed to use crontab syntax to specify intervals.